### ISSUE TITLE
Confirm registration with authorizedCallback

### ISSUE THREAD

**Issue by john-hi** [damienbod/angular-auth-oidc-client#292]
What is the right way to confirm email and redirect to client application already authorized?

Since we come from outside our application we have no local_state and state verification doesn't pass with
```
ValidateStateFromHashCallback failed, state: undefined local_state:
authorizedCallback incorrect state
```
All attempts to pass an empty state have failed (and I guess if it was success it would cause security issue).
Now I have an auto-login component to catch failed authorization result and redirect to. But I think it's not right way.

Any suggestions?

**Comment by profet23** [comment]
Not sure I understand your workflow. 

Applications can only trust 2 things. Themselves (even this is alittle iffy in browser land) and their authority (identity server). 

If you:

> come from outside our application

Then you need to talk to your authority and confirm the identity of the user. You can't simply trust what is passed to you. That's not oidc, thats just token verification. Which is fine... but not really the point of this project.

**Comment by john-hi** [comment]
Yes, I'm coming "from outside our (angular) application", but I'm coming from Identity server via creating redirect callback url by CreateAuthorizeUrl(...). What could be more authoritative?

**Comment by john-hi** [comment]
Firstly, I really appreciate your participation. I'm sorry if i was not clear enough. It is not login flow issue. Login works fine. It is part of "sign up / registration" flow... 

1. User click on Sign Up button.
2. Application redirect him to Identity Server for registration
3. User input his email address and passwords.
4. Identity Server send him confirmation email with hashed link.
5 User goes to Identity Server via link in email.
6. Identity Server confirm email registration, authenticate user and redirect back to application.

Everything above but last point are not a part of this project so was missed. But how application can accept authorized callback if it is not the initiator of authorization? Application has no internal state so authorization proccess has failed. Now I catch this situation and redirect to "autologin-component" to initiate autorization again. Since Identity Server has already authenticated this user, it just redirect him back to application and straight forward to your fifth point... For user it looks like chain splashing screen redirects which get bad experience...

**Comment by profet23** [comment]
@john-hi could you try this with 6.0.12?

Make sure you have silent renew configured correctly and you should no longer have to redirect from your angular app after step 6 (above).

@damienbod  I believe this can be closed...

Also.. awesome how many issues were able to be closed by 6.0.12!
